pkg/tasks/c1api/service_client.go

```diff
@@ -0,0 +1,253 @@
+	if err != nil {
+		return nil, err
+	}
+	defer done()
+
+	in.HostId = c.getHostID()
+
+	return client.Heartbeat(ctx, in)
+}
+
+func (c *c1ServiceClient) FinishTask(ctx context.Context, in *v1.BatonServiceFinishTaskRequest) (*v1.BatonServiceFinishTaskResponse, error) {
+	client, done, err := c.getClientConn(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer done()
+
+	in.HostId = c.getHostID()
+
+	return client.FinishTask(ctx, in)
+}
+
+func (c *c1ServiceClient) Upload(ctx context.Context, task *v1.Task, r io.ReadSeeker) error {
+	l := ctxzap.Extract(ctx)
+
+	client, done, err := c.getClientConn(ctx)
+	if err != nil {
+		return err
+	}
+	defer done()
+
+	uc, err := client.UploadAsset(ctx)
+	if err != nil {
+		return err
+	}
+
+	hasher := sha256.New()
+	rLen, err := io.Copy(hasher, r)
+	if err != nil {
+		l.Error("failed to calculate sha256 of upload asset", zap.Error(err))
+		return err
+	}
+	shaChecksum := hasher.Sum(nil)
+
+	_, err = r.Seek(0, io.SeekStart)
+	if err != nil {
+		l.Error("failed to seek to start of upload asset", zap.Error(err))
+		return err
+	}
+
+	err = uc.Send(&v1.BatonServiceUploadAssetRequest{
+		Msg: &v1.BatonServiceUploadAssetRequest_Metadata{
+			Metadata: &v1.BatonServiceUploadAssetRequest_UploadMetadata{
+				HostId: c.getHostID(),
+				TaskId: task.Id,
+			},
+		},
+	})
+	if err != nil {
+		l.Error("failed to send upload metadata", zap.Error(err))
+		return err
+	}
+
+	chunkCount := uint64(math.Ceil(float64(rLen) / float64(fileChunkSize)))
+	for i := uint64(0); i < chunkCount; i++ {
+		l.Debug("sending upload chunk", zap.Uint64("chunk", i), zap.Uint64("total_chunks", chunkCount))
+
+		chunkSize := fileChunkSize
+		if i == chunkCount-1 {
+			chunkSize = int(rLen) - int(i)*fileChunkSize
+		}
+
+		chunk := make([]byte, chunkSize)
+		_, err = r.Read(chunk)
+		if err != nil {
+			l.Error("failed to read upload asset", zap.Error(err))
+			return err
+		}
+
+		err = uc.Send(&v1.BatonServiceUploadAssetRequest{
+			Msg: &v1.BatonServiceUploadAssetRequest_Data{
+				Data: &v1.BatonServiceUploadAssetRequest_UploadData{
+					Data: chunk,
+				},
+			},
+		})
+		if err != nil {
+			l.Error("failed to send upload chunk", zap.Error(err))
+			return err
+		}
+	}
+
+	err = uc.Send(&v1.BatonServiceUploadAssetRequest{
+		Msg: &v1.BatonServiceUploadAssetRequest_Eof{
+			Eof: &v1.BatonServiceUploadAssetRequest_UploadEOF{
+				Sha256Checksum: shaChecksum,
+			},
+		},
+	})
+	if err != nil {
+		l.Error("failed to send upload metadata", zap.Error(err))
+		return err
+	}
+
+	_, err = uc.CloseAndRecv()
+	if err != nil {
+		l.Error("failed to close upload client", zap.Error(err))
+		return err
+	}
+
+	return nil
+}
+
+func newServiceClient(ctx context.Context, clientID string, clientSecret string) (BatonServiceClient, error) {
+	credProvider, clientName, tokenHost, err := ugrpc.NewC1CredentialProvider(ctx, clientID, clientSecret)
+	if err != nil {
+		return nil, err
+	}
+
+	if envHost, ok := os.LookupEnv("BATON_C1_API_HOST"); ok {
```

Comment:
Should Baton reference C1? Are we releasing a Baton server since it's open source?